superset-frontend/src/utils/localStorageHelpers.ts

```diff
@@ -20,7 +20,10 @@
 export const getFromLocalStorage = (key: string, defaultValue: any) => {
   try {
     const value = localStorage.getItem(key);
-    return JSON.parse(value || 'null') || defaultValue;
+    if (value === null) {
+      return defaultValue;
+    }
+    return JSON.parse(value);
```

Comment:
this was broken in a very subtle way before. If the value in localStorage was falsey (0, false, '', etc.), then this function would return the defaultValue.

unfortunately, localStorage doesn't have a `hasItem` function, so the best thing we can do is see if `getItem` returns null (https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem#return_value). Isn't JS fun!